.changeset/polite-wombats-smash.md

```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/backend-app-api': patch
+'@backstage/errors': patch
+---
+
+Add NotImplementedError
```

Comment:
Let's split this into two changesets, one for errors with a quick blurb about the purpose of the new error, and for backend-app-api we just mention that we add support for it